src/index.js

```diff
@@ -52,16 +52,24 @@ function updateData (chartInstance, callback) {
       y = y > chartInstance.scales[scale].max ? chartInstance.scales[scale].max : y
       y = y < chartInstance.scales[scale].min ? chartInstance.scales[scale].min : y
 
-      if(chartInstance.data.datasets[datasetIndex].data[index].x !== undefined && chartInstance.options.dragX) {
-        chartInstance.data.datasets[datasetIndex].data[index].x = x
-      }
-
-      if(chartInstance.data.datasets[datasetIndex].data[index].y !== undefined) {
-        chartInstance.data.datasets[datasetIndex].data[index].y = y
-      }
-      else {
-        chartInstance.data.datasets[datasetIndex].data[index] = y
-      }
+if (chartInstance.data.datasets[datasetIndex].data[index].x !== undefined && chartInstance.options.dragX) {
+	        chartInstance.data.datasets[datasetIndex].data[index] = {
+						...chartInstance.data.datasets[datasetIndex].data[index],
```

Comment:
The `...` gives me a syntax error at this position. Could it be that you were using `TypeScript`/`Flow` or any of the likes?

Reply:
That's all fine, and I absolutely believe you that this fixes your issue. But this project needs to compile as a stand-alone as well. This is especially important for users who directly import the module via `<script>` tags. You can check this if you run `npm run compile` on your own fork of this module. It could be that you need to adjust the Webpack config in this project to cope with your changes. 
Again, I think it would be really helpful if you could share a working example where this section of the code breaks functionality.

Reply:
Right now i cant give you a working example, because its a big project and he under the NDA, but on this weekend i make a project on jsfiddle/codepen and give you link here.

Reply:
Ok, thanks a lot for your efforts!